connector.go

```diff
@@ -5,30 +5,37 @@ import (
 	"go.arcalot.io/log/v2"
 	"go.flow.arcalot.io/deployer"
 	"go.flow.arcalot.io/pythondeployer/internal/cliwrapper"
-	"os"
+	"sync"
 )
 
 type Connector struct {
-	config           *Config
-	logger           log.Logger
-	pythonCliWrapper cliwrapper.CliWrapper
+	config        *Config
+	logger        log.Logger
+	pythonFactory cliwrapper.CliWrapperFactory
+	venvs         map[string]struct{}
```

Comment:
I'm kinda confused here. I only see a couple of references to this `venvs` map, and none actually seem to use it as a map. At `connector.go` 52 you check whether the *key* `fullModuleName` exists in the map, while at `connector.go` 58 and `factory.go` 58 you assign an empty struct to that key.

Maybe I'm missing it: but a comment here describing the intended purpose wouldn't be wasted, because it's not obvious. 😦

Reply:
Golang does not have a `set` type built into the standard library, so this is the current convention to create and use sets, without creating a bunch of OOP accessors and getters.

Reply:
That's kinda hilarious, actually. A map with an unused value to implement a hashed key lookup. Yeah, OK, vaguely like set semantics, and if it's a common Go hack maybe it doesn't require a comment. (But, why a struct instead of a scalar value type?)